napi_from_js.md

```diff
@@ -0,0 +1,367 @@
+```js
+function AddTwo(num)
+{
+    return num + 2;
+}
+```
+
+Then, the above function can be invoked from your native add-on using the
+following code:
+```C
+// Get the function named "AddTwo" on the global object
+napi_value global, add_two, arg;
+napi_status status = napi_get_global(env, &global);
+if (status != napi_ok) return;
+
+status = napi_get_named_property(env, global, "AddTwo", &add_two);
+if (status != napi_ok) return;
+
+// var arg = 1337
+status = napi_create_number(env, 1337, &arg);
+if (status != napi_ok) return;
+
+napi_value* argv = &arg;
+size_t argc = 1;
+
+// AddTwo(arg);
+napi_value return_val;
+status = napi_call_function(env, global, add_two, argc, argv, &return_val);
+if (status != napi_ok) return;
+
+// Convert the result back to a native type
+int32_t result;
+status = napi_get_value_int32(env, return_val, &result);
+if (status != napi_ok) return;
+```
+
+### *napi_create_function*
+
+#### Signature
+```C
+napi_status napi_create_function(napi_env env,
+                                 const char* utf8name,
+                                 napi_callback cb,
+                                 void* data,
+                                 napi_value* result);
+```
+
+#### Parameters
+- `[in]  env `: The environment that the API is invoked under
+- `[in]  utf8Name`: The name of the function, encoded as UTF8. This is visible
+within JavaScript as the new function object's `name` property.
+- `[in]  cb`: The native function which should be called when this function 
+object is invoked
+- `[in]  data`: User-provided data context. This will be passed back into the 
+function when invoked later.
+- `[out] result`: `napi_value` representing the JavaScript function object for
+the newly created function
+
+#### Description
+This API allows an add-on author to create a function object in native code. 
+This is the primary mechanism to allow calling *into* the add-on's native code
+*from* Javascript.
+
+Note however, the newly created function is not automatically  visible from
+script after this call. Instead, you must explicitly set a property on the
+Global object (or any other object that is visible to JavaScript), in order for
+the function to be accessible from script. For instance, if you wish to expose a
+function as part of the add-on's module exports, you can set the newly created
+function on the exports object. A sample module might look as follows:
+```C
+void SayHello(napi_env env, napi_callback_info info) {
+  printf("Hello\n");
+}
+
+void Init(napi_env env, napi_value exports, napi_value module, void* priv) {
+  napi_status status;
+
+  napi_value fn;
+  status =  napi_create_function(env, NULL, SayHello, NULL, &fn);
+  if (status != napi_ok) return;
+
+  status = napi_set_named_property(env, exports, "sayHello", fn);
+  if (status != napi_ok) return;
+}
+
+NAPI_MODULE(add-on, Init)
+```
+
+Given the above code, the add-on can be used from JavaScript as follows:
+```js
+var myaddon = require('myadd-on');
```

Comment:
The final value of the string passed to require is actually not that which was specified in `NAPI_MODULE()` but the name of the target in `binding.gyp` responsible for creating the `.node` file. It must also be a path that starts with `./` or `/`, otherwise Node will start searching for a package named "myadd-on" rather than loading a file at a specific path.